### PR TITLE
Use job name improve CWL tool debugging (connected to #1343)

### PR DESCRIPTION
**Pull request by cket**

### CODE REVIEW

**Comment by cket** [comment]
jenkins, test this please

**Comment by cket** [comment]
@chapmanb @mr-c would either of you mind taking a look at how I'm attempting to label CWL jobs? Any suggestions on which CWL tool attributes to use for naming would be appreciated!

**Comment by chapmanb** [comment]
Christopher;
Thanks again for taking this on. This looks good for the jobName. I'd suggest using the executed command for the other debugging slot:
```
self.unitName = " ".join(cwltool.tool.get("baseCommand", []))
```
We don't have access to sample information at this point since it we're dealing with the workflow definition (in CWL) not the samples that are passing through, so don't have a way to set that. If Toil somehow knows to set unitName correctly we could instead concatenate the command information to the `id` and leave sample setting in unitName as is.

Thanks again for doing this. Once you get it integrated I'll work on rebasing it against my cwltool branch update.

**Comment by cket** [comment]
jenkins, test this please

**Comment by cket** [comment]
jenkins, test this please

**Comment by cket** [comment]
@chapmanb it now handles both strings & lists for the baseCommand attribute - do you want to take another look before I merge it?

File: src/toil/cwl/cwltoil.py [hunk 1]
```diff
@@ -221,12 +221,22 @@ def __init__(self, tool, cwljob, **kwargs):
         builder.timeout = 0
         builder.resources = {}
         req = tool.evalResources(builder, {})
+        self.cwltool = tool
+        # pass the default of None if basecommand is empty
+        unitName = self.cwltool.tool.get("baseCommand", None)
+        if isinstance(unitName, list):
```
Comment (chapmanb) [inline src/toil/cwl/cwltoil.py]: You might want `isinstance(unitName, (list, tuple)):` here, just in case.

Comment (cket) [inline src/toil/cwl/cwltoil.py]: Good catch, thanks.

**Comment by cket** [comment]
@chapmanb thanks, I'll make that change & we should have this merged today!

**Pull request merged**